File: app/src/main/java/net/felipemuniz/personmvvm/domain/entities/Person.java
/**
 * Copyright 2016 Erik Jhordan Rey.
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.felipemuniz.personmvvm.domain.entities;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

/**
 * Created by Muniz on 05/07/2017.
 */


@Entity
@Parcel
public class Person {

  @PrimaryKey(autoGenerate = true)
  public int id;

  @SerializedName("name")
  private String name;

  @SerializedName("surname")
  private String surname;

  @SerializedName("gender")
  private String gender;

  @SerializedName("region")
  private String region;

  @SerializedName("age")
  private String age;

  @SerializedName("phone")
  private String phone;

  @SerializedName("photo")
  private String photo;

  public String getFullName() {
    return name + " " + surname;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getSurname() {
    return surname;
  }

  public void setSurname(String surname) {
    this.surname = surname;
  }

  public String getGender() {
    return gender;
  }

  public void setGender(String gender) {
    this.gender = gender;
  }

  public String getRegion() {
    return region;
  }

  public void setRegion(String region) {
    this.region = region;
  }

  public String getAge() {
    return age;
  }

  public void setAge(String age) {
    this.age = age;
  }

  public String getPhone() {
    return phone;
  }

  public void setPhone(String phone) {
    this.phone = phone;
  }

  public String getPhoto() {
    return photo;
  }

  public void setPhoto(String photo) {
    this.photo = photo;
  }
}

File: app/src/main/java/net/felipemuniz/personmvvm/domain/entities/Person.kt
/**
 * Copyright 2016 Erik Jhordan Rey.
 *
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.felipemuniz.personmvvm.domain.entities

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import com.google.gson.annotations.SerializedName

import org.parceler.Parcel

/**
 * Created by Muniz on 05/07/2017.
 */


@Entity
@Parcel
class Person {

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    @SerializedName("name")
    var name: String? = null

    @SerializedName("surname")
    var surname: String? = null

    @SerializedName("gender")
    var gender: String? = null

    @SerializedName("region")
    var region: String? = null

    @SerializedName("age")
    var age: String? = null

    @SerializedName("phone")
    var phone: String? = null

    @SerializedName("photo")
    var photo: String? = null

    val fullName: String
        get() = name + " " + surname
}
